Include/internal/pycore_regions.h

```diff
@@ -83,6 +86,14 @@ int _PyRegion_is_closed(PyObject* region);
 int _PyCown_release(PyObject *self);
 int _PyCown_is_released(PyObject *self);
 
+
+#define Py_CLEAR_OBJECT_FIELD(obj, field) do { \
+    PyObject *tmp = (PyObject *)(field); \
+    (field) = NULL; \
+    Py_REGIONREMOVEREFERENCE(obj, tmp); \
+    Py_XDECREF(tmp); \
```

Comment:
I wonder if we should split this macro into an `X` (can be null) and non-`X` variant? :thinking: It's easier to do them now as separate macros then adding this separation afterwards

Reply:
Agreed. I can do that

Reply:
So there isn't a `Py_XCLEAR`.  So I am less sure about doing this now.  I'm going to improve the macro inline with `Py_CLEAR`, but leave this for now.